static/absolute/js/searchPage.js

```diff
@@ -101,19 +99,20 @@ socket.on("refresh", () => {
   console.log(
     "Refresh message received from server via Socket.io. Refreshing restaurant results."
   );
-  searchRestaurants();
+  refreshUI();
 });
 
-async function searchRestaurants() {
+async function refreshUI() {
   const eventId = getEventId();
-  const response = await (await fetch(`/api/${eventId}/restaurants`)).json();
-  if (response.status !== 200) {
-    console.log("Error fetching restaurants: " + response.error);
-    return;
-  }
-
-  initMap();
-  showRestaurants(response.data);
+  const participantsResponse = await (
```

Comment:
my knowledge on promises is not great, but does this imply the two fetch events here are not being run simultaneously? if so, can they be run simultaneously instead to save some time?

Reply:
In fact, could showRestaurants() even be made async so that it can start running while the participants fetch is still ongoing?

Reply:
@ChisomOkwor this can be done using `Promise.all` -- https://stackoverflow.com/questions/59037553/fetch-multiple-urls-at-the-same-time

Reply:
could this even be done with something like this?

const participantsPromise = (await fetch(`/api/${eventId}/participants`)).json()
const restaurantsPromise = (await fetch(`/api/${eventId}/restaurants`)).json();
showRestaurants(await restaurantsPromise);
initMap(await participantsPromise, await restaurantsPromise);

Or does JS not allow that syntax?